src/adapter.ts

```diff
@@ -26,9 +32,42 @@ export interface TableRow {
 export abstract class DbAdapter {
     public abstract init(): Promise<void>
     public abstract getLastMigrationTask(): Promise<Task | null>
-    public abstract logMigrationTask(task: Task): Promise<void>
+    public abstract beginMigrationTask(task: Task): Promise<void>
+    public abstract finishMigrationTask(task: Task): Promise<void>
     public abstract checkIfTaskCanExecute(task: Task): Promise<void>
     public abstract close(): Promise<void>
+    protected abstract hasPendingMigration(): Promise<boolean>
+
+    public waitForPending(logger: Logger): Promise<boolean> {
+        let wasPending = false
+        return new Promise<boolean>(async (resolve, reject) => {
+            // fail after 10 min
+            const timeout = setTimeout(() => reject(new PendingMigrationTimedOutError()), 1000 * 60 * 10)
```

Comment:
If you just want to fail the Promise after 10min, I would use `Promise.race()` to race the timeout Promise against the operation Promise